src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotificationBundle.java

```diff
@@ -0,0 +1,62 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import org.kohsuke.github.GHCommitState;
+
+public class GitHubBuildStatusNotificationBundle {
+
+    private final String context;
+    private final String url;
+    private final String message;
+    private final GHCommitState state;
+
+    public GitHubBuildStatusNotificationBundle(String context, String url, String message, GHCommitState state) {
```

Comment:
Probably better to use a `GitHubNotificationDetails.Builder` pattern to build instances. That will also allow you to validate in the `build()` method